File: frontend/src/services/carritoService.js
// src/services/carritoService.js
const API_URL = process.env.REACT_APP_API_URL || "http://localhost:8081";

class CarritoService {
  /**
   * Agregar producto al carrito en el backend
   */
  async agregarProducto(productoId, cantidad, comentarios = "") {
    try {
      const response = await fetch(`${API_URL}/api/carrito/agregar`, {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        credentials: "include", // Para enviar cookies de sesión
        body: JSON.stringify({
          productoId,
          cantidad,
          comentarios,
        }),
      });

      if (!response.ok) {
        throw new Error(`Error HTTP: ${response.status}`);
      }

      return await response.json();
    } catch (error) {
      console.error("Error al agregar producto al carrito:", error);
      throw error;
    }
  }

  /**
   * Actualizar cantidad de un producto
   */
  async actualizarCantidad(productoId, cantidad) {
    try {
      const response = await fetch(
        `${API_URL}/api/carrito/actualizar/${productoId}`,
        {
          method: "PUT",
          headers: {
            "Content-Type": "application/json",
          },
          credentials: "include",
          body: JSON.stringify({ cantidad }),
        }
      );

      if (!response.ok) {
        throw new Error(`Error HTTP: ${response.status}`);
      }

      return await response.json();
    } catch (error) {
      console.error("Error al actualizar cantidad:", error);
      throw error;
    }
  }

  /**
   * Eliminar producto del carrito
   */
  async eliminarProducto(productoId) {
    try {
      const response = await fetch(
        `${API_URL}/api/carrito/eliminar/${productoId}`,
        {
          method: "DELETE",
          credentials: "include",
        }
      );

      if (!response.ok) {
        throw new Error(`Error HTTP: ${response.status}`);
      }

      return await response.json();
    } catch (error) {
      console.error("Error al eliminar producto:", error);
      throw error;
    }
  }

  /**
   * Obtener carrito actual
   */
  async obtenerCarrito() {
    try {
      const response = await fetch(`${API_URL}/api/carrito`, {
        method: "GET",
        credentials: "include",
      });

      if (!response.ok) {
        throw new Error(`Error HTTP: ${response.status}`);
      }

      return await response.json();
    } catch (error) {
      console.error("Error al obtener carrito:", error);
      throw error;
    }
  }

  /**
   * Limpiar carrito
   */
  async limpiarCarrito() {
    try {
      const response = await fetch(`${API_URL}/api/carrito/limpiar`, {
        method: "DELETE",
        credentials: "include",
      });

      if (!response.ok) {
        throw new Error(`Error HTTP: ${response.status}`);
      }

      return await response.json();
    } catch (error) {
      console.error("Error al limpiar carrito:", error);
      throw error;
    }
  }

  /**
   * Obtener total de items
   */
  async obtenerTotalItems() {
    try {
      const response = await fetch(`${API_URL}/api/carrito/total-items`, {
        method: "GET",
        credentials: "include",
      });

      if (!response.ok) {
        throw new Error(`Error HTTP: ${response.status}`);
      }

      const data = await response.json();
      return data.totalItems;
    } catch (error) {
      console.error("Error al obtener total de items:", error);
      throw error;
    }
  }
}

export default new CarritoService();

File: backups/Full_Backup_2025-07-16_19-12-38/source/backend/src/main/java/com/comoencasa_backend/service/CarritoService.java
package com.comoencasa_backend.service;

import com.comoencasa_backend.dto.CarritoDTO;
import com.comoencasa_backend.dto.CarritoItemDTO;

public interface CarritoService {
    
    /**
     * Agregar producto al carrito
     * @param sessionId ID de sesión del cliente
     * @param productoId ID del producto
     * @param cantidad Cantidad a agregar
     * @param comentarios Comentarios especiales
     * @return Carrito actualizado
     */
    CarritoDTO agregarProducto(String sessionId, Long productoId, Integer cantidad, String comentarios);

    /**
     * Actualizar cantidad de un producto en el carrito
     * @param sessionId ID de sesión del cliente
     * @param productoId ID del producto
     * @param nuevaCantidad Nueva cantidad
     * @return Carrito actualizado
     */
    CarritoDTO actualizarCantidad(String sessionId, Long productoId, Integer nuevaCantidad);

    /**
     * Eliminar producto del carrito
     * @param sessionId ID de sesión del cliente
     * @param productoId ID del producto a eliminar
     * @return Carrito actualizado
     */
    CarritoDTO eliminarProducto(String sessionId, Long productoId);

    /**
     * Obtener carrito por sesión
     * @param sessionId ID de sesión del cliente
     * @return Carrito del cliente
     */
    CarritoDTO obtenerCarrito(String sessionId);

    /**
     * Limpiar carrito
     * @param sessionId ID de sesión del cliente
     * @return Carrito vacío
     */
    CarritoDTO limpiarCarrito(String sessionId);

    /**
     * Obtener total de items en el carrito
     * @param sessionId ID de sesión del cliente
     * @return Número total de items
     */
    Integer obtenerTotalItems(String sessionId);
}
